### ISSUE TITLE
[HELP] Zulfbungee reporting a SocketException on a Paper server

### ISSUE THREAD

**Issue by CerialPvP** [Zulfen/ZulfBungee#18]
I'm not sure if this is something on my end, or a bug on the plugin, but here we are.
When I start my paper & velocity servers (i have debug mode on), my velocity server does not report any errors, but the paper servers reports the following:
```
06.03 03:49:17 [Server] [INFO] [ZulfBungee] Error while creating socket: java.net.SocketException: Invalid argument
```
Velocity config:
```yml
# - Key Setup -



# This is the port that this server will bind to, which will it listen for incoming connections on.

port: 8077



# The IP that this server will bind to. Similar to the "client-host" in the Spigot config.

# This should usually be left as 0.0.0.0, or 127.0.0.1, but feel free to change this if you need to -

# (i.e. you are on shared hosting / Pterodactyl)

host: "127.0.0.1"



# - Options -



# This will toggle the display of detailed messages about what is going to the console.

# When filing a bug report / checking for issues, this is extremely useful to know what exactly went wrong!

debug: true



# Enables or disables global scripts.

global-scripts: true



# - Storage -



# What kind of database should be used to store network variables.

# Currently, supported databases: MySQL, H2



# If you have no clue what this does, just leave this as the default.

storage-type: "h2"



# Each option should be self-explanatory if you know what you are doing.



# If you are using H2, this will not apply - these two settings only apply to remote databases.

storage-host: "localhost"

storage-port: 3306



# !! Change these from the default to improve security !!

storage-username: "username"

storage-password: "password"



storage-database: "database"



# [MySQL Setup]



# If you get errors in console because of SSL, disable it here, but it's better to set up SSL on the MySQL server.

mysql-use-ssl: true

mysql-verify-certificate: true
```

Paper config (each server has a different port, I made sure to open them):
```yml
# - Key Setup -



# The port which you defined in your proxy config.

server-port: 8077



# The IP of your proxy.

server-host: "51.222.208.87"



# The port which this client (this Spigot server) will bind to, so it can listen for the proxy.

client-port: 8160



# The IP this client will bind to.

# This should usually be left as 0.0.0.0, or 127.0.0.1, but feel free to change this if you need to -

# (i.e. you are on shared hosting / Pterodactyl)

client-host: "127.0.0.1"



# If your proxy cannot find the name of your server from



# - Options -



# How many milliseconds it will wait for a connection.

# A timeout will be pronounced if the server does not respond in this time.

connection-timeout: 6000





# This will toggle the display of detailed messages about what is going to the console.

# When filing a bug report / checking for issues, this is extremely useful to know what exactly went wrong!

debug: true
```
It was on 0.0.0.0, and it gave a different error, if you want it ill send it here.

**Comment by CerialPvP** [comment]
@Zulfen

**Comment by Zulfen** [comment]
Hi! Just wanna check - what kind of environment are you running in? Is this Pterodactyl, a VPS or you hosting this locally?
Are the Paper servers on a different machine to your proxy?

**Comment by CerialPvP** [comment]
> Hi! Just wanna check - what kind of environment are you running in? Is this Pterodactyl, a VPS or you hosting this locally on your PC? Are the Paper servers on a different machine to your proxy?

I'm running this on pebblehost (shared hosting plan), which doesn't run on Pterodactyl on my knowledge. Today I will do some testing locally, but if you can help me, that would be 100 times better.

**Comment by CerialPvP** [comment]
And yes, every single one is on a different machine. There is one server with the same node as another one, but that is all on the terms of "same machine".

**Comment by Zulfen** [comment]
Ah, I see.
Try changing the `host` in ZulfBungee's Velocity config to `0.0.0.0`, and your `client-host` in each Spigot server to `0.0.0.0` also. 
Make sure all those ports are open (but I think all the ports you get from PebbleHost are automatically forwarded, so no need to worry).

**Comment by CerialPvP** [comment]
Okay, lemme try that.

**Comment by CerialPvP** [comment]
![image](https://user-images.githubusercontent.com/71380797/223696501-a13fcd2d-9d8f-4bac-89d8-524fa22c9241.png)
Now I get a completely different error.

**Comment by CerialPvP** [comment]
![image](https://user-images.githubusercontent.com/71380797/223696646-6996db3f-f8da-45a9-a60a-77a824392d69.png)
This is in the proxy.

**Comment by CerialPvP** [comment]
![image](https://user-images.githubusercontent.com/71380797/223701153-7384e511-f41e-4a71-be54-2f8027d026fd.png)
It only detects one server.

**Comment by CerialPvP** [comment]
@Zulfen hello?

**Comment by CerialPvP** [comment]
@Zulfen It's really urgent can you help? And can I have your discord so we can communicate faster?

**Comment by CerialPvP** [comment]
It has been 3 days wya?

**Comment by Zulfen** [comment]
Hi! Sorry to keep you waiting, this bug should be fixed with the latest update, let me know if it works!

**Comment by Zulfen** [comment]
Sorry to keep you waiting!
Try setting your `client-host` on each Spigot server to `0.0.0.0`. Also check if those ports are open :)

**Comment by CerialPvP** [comment]
I'll try that when I'll be home. Can you give me your Discord so we can get in touch faster? If you need mine it's Cerial#7773

**Comment by CerialPvP** [comment]
Well according to your first comment you already told me to set the IP to 0.0.0.0, which I already did.

The ports are also open like they should, I'll check it out more at home since I'm at school right now.

**Comment by CerialPvP** [comment]
It has been a week... I have already done that because you told me to do it here:

> Ah, I see.
> Try changing the `host` in ZulfBungee's Velocity config to `0.0.0.0`, and your `client-host` in each Spigot server to `0.0.0.0` also.
> Make sure all those ports are open (but I think all the ports you get from PebbleHost are automatically forwarded, so no need to worry).

I'm sorry if I'm bothering you too much, but we really need this to work. If you need me to provide you with something, tell me. And can we please get in touch on Discord so we both don't have to wait for replies from each other for weeks on end?

**Comment by Zulfen** [comment]
My discord is zulfen#3004

**Comment by CerialPvP** [comment]
Sorry for the late reply. I have sent you a friend request, my Discord is Cerial#7773.

**Comment by Zulfen** [comment]
Issue resolved with latest feature in v0.9.5.

**Comment by CerialPvP** [comment]
hey, sorry for suddenly replying in this issue, my main got disabled for underage (even tho im not) and i sent u a friend request from my alt Cerial2#5420
